### PR TITLE
1.3 relnotes

### PR DESCRIPTION
**Pull request by brson**
I'd love to have any tips about highlights and lang stuff I missed. Sadly, this needs to be merged _tomorrow_.

[Rendered](https://github.com/brson/rust/blob/relnotes/RELEASES.md)

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,5 +1,128 @@
...
+  [`Any`] trait.
+* The [`contains`] method for searching strings now [employes the
+  two-way algorithm][search] instead of doing a naive search, for
+  significant speedups.
+* The performance of `PartialEq` for slices is [much faster][ps].
+* The [`Hash`] trait offers the default method, [`hash_slice`], which
+  is overridden and optimized by the implementations for scalars.
+* The [`Hasher`] trait now has a number of specialized `write_*`
+  methods for primitive types, for efficiency.
+* The I/O-specific error type, [`std::io::Error`][ie], gained a set of
```
Comment (sfackler) [inline RELEASES.md]: `io::Error::cause` also now forwards to the inner `Error`'s.

File: RELEASES.md [hunk 2]
```diff
@@ -1,5 +1,128 @@
...
+* The owned C string, [`CString`], implements [`Borrow`] and the
+  borrowed C string, [`CStr`], implements [`ToOwned`]. The two of
+  these allow C strings to be borrowed and cloned in generic code.
+* [`CStr`] implements [`Debug`].
+* [`AtomicPtr`] implements [`Debug`].
+* [`Error`] trait objects [can be downcast to their concrete types][e]
+  in many common configurations, using the [`is`], [`downcast`],
+  [`downcast_ref`] and [`downcast_mut`] methods, similarly to the
+  [`Any`] trait.
+* The [`contains`] method for searching strings now [employes the
```
Comment (Munksgaard) [inline RELEASES.md]: `employes` should probably be `employs`.

Comment (Munksgaard) [inline RELEASES.md]: And the `contains` link doesn't seem to work for me for some reason.

Comment (bluss) [inline RELEASES.md]: It's not just `contains`, it's all `&str` in `&str` search that uses the two way algorithm, that includes methods like `contains, find, rfind, split, matches, trim_matches` etc.

`starts_with` / `ends_with` were also improved in that PR.

File: RELEASES.md [hunk 3]
```diff
@@ -1,5 +1,128 @@
...
+* The performance of `PartialEq` for slices is [much faster][ps].
+* The [`Hash`] trait offers the default method, [`hash_slice`], which
+  is overridden and optimized by the implementations for scalars.
+* The [`Hasher`] trait now has a number of specialized `write_*`
+  methods for primitive types, for efficiency.
+* The I/O-specific error type, [`std::io::Error`][ie], gained a set of
+  methods for accessing the 'inner error', if any: [`get_ref`][iegr],
+  [`get_mut`][iegm], [`into_inner`][ieii].
+* [`process::Child`][pc] gained the [`id`] method, which returns a
+  `u32` representing the platform-specific process handle.
```
Comment (retep998) [inline RELEASES.md]: Not a handle! Just a process identifier. There is also an associated handle for each process which is exposed using `AsRawHandle`.

File: RELEASES.md [hunk 4]
```diff
@@ -1,5 +1,128 @@
...
+* [The Rustonomicon][nom] is a new book in the official documentation
+  that dives into writing unsafe Rust.
+* The [`Duration`] API, [has been stabilized][ds]. This basic unit of
+  timekeeping is employed by other std APIs, as well as out-of-tree
+  time crates.
+
+Breaking Changes
+----------------
+
+* The `#[prelude_import]`, an internal implementation detail, was
```
Comment (bluss) [inline RELEASES.md]: I think there's a missing word, "The `#[prelude_import]` *attribute*",

**Comment by arielb1** [comment]
[`unused_mut`](#26378) and [`unconditional_recursion`](#26783) became more strict.

**Comment by arielb1** [comment]
[with landing pads disabled, panics kill the process instead of leaking the task](c35b2bd226736925961ca6853b2ef29e8094cd90)

**Comment by arielb1** [comment]
[the new object lifetime defaults are in effect](https://github.com/rust-lang/rust/27045)

**Comment by arielb1** [comment]
[the lifetime elision rules changed](https://github.com/rust-lang/rust/issues/27248)

**Comment by brson** [comment]
Woohoo! Thanks for the review folks. This helps a lot.

**Comment by bluss** [comment]
I think  #26241 is in 1.3 as well? It allows a slew of mutable functionality to String and `&mut str`. In the end, the library functionality it enables is rather limited, but you can now produce `&mut str` easily through slicing.

**Comment by brson** [comment]
@bluss I you are right. I didn't include it because I thought it wasn't stable, but it's 'insta-stable'.

**Comment by brson** [comment]
Some more from this timeframe with the relnotes tag: https://github.com/search?utf8=%E2%9C%93&q=org%3Arust-lang+updated%3A2015-06-25..2015-08-06+is%3Amerged+label%3Arelnotes&type=Issues&ref=searchresults

I'll make sure all these are in.

**Comment by brson** [comment]
r? @aturon 

Last call. I will back port these in an hour.

**Comment by aturon** [comment]
@brson: LGTM

File: RELEASES.md [hunk 5]
```diff
@@ -1,5 +1,211 @@
...
+[sh]: https://github.com/rust-lang/rust/pull/27280
+[srf]: http://doc.rust-lang.org/nightly/std/primitive.str.html#method.rfind
+[ss]: http://doc.rust-lang.org/nightly/std/primitive.str.html#method.split
+[ssw]: http://doc.rust-lang.org/nightly/std/primitive.str.html#method.starts_with
+[st]: https://github.com/rust-lang/rust/pull/26630
+[win1]: https://github.com/rust-lang/rust/pull/26569
+[win2]: https://github.com/rust-lang/rust/pull/26741
+[win3]: https://github.com/rust-lang/rust/pull/26741
+[win4]: https://github.com/rust-lang/rust/pull/27210
+[xp]: https://github.com/rust-lang/rust/pull/26569
```
Comment (SimonSapin) [inline RELEASES.md]: Was this meant to be https://github.com/rust-lang/rust/pull/26601 instead?

**Pull request merged**